client/consensus/src/collators/lookahead.rs

```diff
@@ -342,6 +613,34 @@ where
                             Ok(h) => h,
                         };
 
+                        // TODO: Currently we use just the first core here, but for elastic scaling
+                        // we iterate and build on all of the cores returned.
+                        // More info: https://github.com/paritytech/polkadot-sdk/issues/1829
+                        let (is_parachain, core_index) = match (&aux_data.slot_freq, core_indices.first()) {
+                            (None, None) => break, // We are parachain and we do not have core allocated, nothing to do,
+                            (None, Some(core_index)) => (true, core_index), // We are parachain and we have core allocated, let's continue
+
+                            (Some(_slot_frequency), None) => {
+                                // We are parathread and core is not allocated. Let's try to buy core
+                                let slot = inherent_providers.slot();
+                                let container_chain_slot_duration = (params.get_current_slot_duration)(parent_header.hash());
+
+                                match try_to_buy_core::<_, _, <<OBlock as BlockT>::Header as HeaderT>::Number, _, CIDP, _>(params.para_id, aux_data, inherent_providers, &params.keystore, orchestrator_client.clone(), orchestrator_tx_pool.clone(), parent_header, params.orchestrator_slot_duration, container_chain_slot_duration).await {
+                                    Ok(block_hash) => {
+                                        tracing::trace!(target: crate::LOG_TARGET, ?block_hash, "Sent unsigned extrinsic to buy the core");
+                                    },
+                                    Err(buy_core_error) => {
+                                        buy_core_error.log_error(slot, params.para_id, relay_parent);
+                                    }
+                                };
+                                break; // No point in continuing as we need to wait for few relay blocks in order for our core to be available.
+                            },
+                            (Some(_slot_frequency), Some(core_index)) => {
+                                tracing::error!(target: crate::LOG_TARGET, ?core_index, "Not a parachain but we have a core (most likely from previous buy order)");
```

Comment:
I would not call this an error, as it sound like something that can happen

Reply:
I guess here we propose right?

Reply:
yes. Will convert it to trace.